code/drasil-code/Language/Drasil/Code/Imperative/Build/Import.hs

```diff
@@ -8,27 +8,30 @@ import Language.Drasil.Code.Imperative.Build.AST (BuildConfig(BuildConfig),
   Runnable(Runnable), BuildName(..), RunType(..))
 import Language.Drasil.Code.Imperative.GOOL.LanguageRenderer (doxConfigName)
 
-import GOOL.Drasil (FileData(..), isSource, isHeader, ModData(..), ProgData(..))
+import GOOL.Drasil (FileData(..), isHeader, ModData(..), ProgData(..), 
+  GOOLState(..), initialState, sources)
 
 import Build.Drasil ((+:+), genMake, makeS, MakeString, mkFile, mkRule,
   mkCheckedCommand, mkFreeVar, RuleTransformer(makeRule))
 
 import Control.Applicative (liftA2)
+import Control.Monad.State (State, evalState, execState)
+import Control.Lens ((^.))
 import Data.Maybe (maybe, maybeToList)
 import System.FilePath.Posix (takeExtension, takeBaseName)
 import Text.PrettyPrint.HughesPJ (Doc)
 
-data CodeHarness = Ch (Maybe BuildConfig) Runnable ProgData [Comments]
+data CodeHarness = Ch (Maybe BuildConfig) Runnable ProgData GOOLState [Comments]
```

Comment:
You should make `CodeHarness` into a record instead of a (named) 5-tuple.  Then you can use either pattern-matching or accessors, whenever each makes the code cleaner.  I won't make this a blocking issue on merge though.